includes/class-newspack-popups-inserter.php

```diff
@@ -660,21 +660,15 @@ public static function assess_categories_filter( $popup ) {
 	 * @return bool Should popup be shown based on tags it has.
 	 */
 	public static function assess_tags_filter( $popup ) {
-		$post_tags = get_the_tags();
-		if ( false === $post_tags ) {
-			$post_tags = [];
-		}
 		$popup_tags = get_the_tags( $popup['id'] );
 		if ( false === $popup_tags ) {
-			$popup_tags = [];
-		}
-		if ( count( $post_tags ) || count( $popup_tags ) ) {
-			return array_intersect(
-				array_column( $post_tags, 'term_id' ),
-				array_column( $popup_tags, 'term_id' )
-			);
+			return true; // No tags on the popup, no need to compare.
```

Comment:
This works well, but a nit: isn't this more or less the same logic as `assess_categories_filter` right above? It'd be nice to standardize both functions so that it's clearer they do the same thing but with different taxonomies. Perhaps we could even refactor both functions into a single function that takes a taxonomy argument to make it extra clear.

Reply:
Decided to move this to a [new PR](https://github.com/Automattic/newspack-popups/pull/518) since the fix is somewhat urgent but the refactor certainly isn't.